### ISSUE TITLE
New angularfire2 version 4.0.0-rc.0 FirebaseObjectObservable with [(ngModel)] binding fails

### ISSUE THREAD

**Issue by Sun3** [angular/angularfire2#943]
I upgraded the Angular 4 app to work with the new AngularFire version 4.0.0-rc.0 and have run into an issue when using <input> control with 2 way data binding [(ngModel)]="contact.name" which now gives new below error. Previous version of AngularFire did not have any errors and worked fine. Now all of the <input> controls do not show data

```
// Uncaught (in promise): TypeError: undefined is not an object (evaluating 'co.contact.name') 

// html
<input type="text" [(ngModel)]="contact.name">
<input type="text" [(ngModel)]="contactSnapshot.name">

// Typescript

contact: FirebaseObjectObservable<any>;
contactSnapshot: any;

constructor(public db: AngularFireDatabase ) {
// Retrieve contact
this.contact = this.db.object('/contacts/-JeBucdFZJIg1Dkio7eN');

// Also via snapshot
this.contact.subscribe(contactSnapshot => {
            this. contactSnapshot = contactSnapshot;
            console.log('contactSnapshot ', this.contactSnapshot);
            console.log('contactSnapshot ', this.contactSnapshot['name']);
});

}
```

### Version info
**Angular:** 4.0.0
**Firebase:** 3.9.0
**AngularFire:** 4.0.0-rc.0

**Comment by davideast** [comment]
@Sun3 Can you provide a plnkr so it's easier to debug? We have a template in the README.

**Comment by WebAddict** [comment]
@davideast your plnkr template doesn't work, hasn't been updated in some time, probably breaking changes of new AF4 is issue. Plnkr error is: "_Can't resolve all parameters for AppComponent_" You should update it to a working version with new AF4.

**Comment by davideast** [comment]
@WebAddict I think it's just a version issue, which is easily updated in SystemJS config. Would you like to send a PR with the updated template?

**Comment by Sun3** [comment]
@davideast Should I wait for the plnkr template to be updated first or?...

**Comment by davideast** [comment]
@Sun3 Zips are hard to deal with. A plnkr is the best solution, I think you just need to update the version numbers in the SystemJS config. Give that a try and if it blows up, let me know I'll take a look.

**Comment by Sun3** [comment]
@davideast This is the plnkr test I setup at [](http://plnkr.co/edit/WRv5kxPaJOmY2gdMLgwf?p=preview)  It should be using the version 4.0.0.rc-0 but please double check. With previous version it worked ok.

In the app.module the real Firebase Credentials would need to be changed.

Thanks.

**Comment by Sun3** [comment]
ps: Not sure if a bit more of the error helps but I place the top portion below. Of course in my actual app I have about 40-50 fields that use two way binding.

```
 Error: Uncaught (in promise): TypeError: undefined is not an object (evaluating 'co.contactSnapshot.name')

checkAndUpdateView@http://localhost:4200/vendor.bundle.js:12256:30
callViewAction@http://localhost:4200/vendor.bundle.js:12571:35
execComponentViewsAction@http://localhost:4200/vendor.bundle.js:12517:27
checkAndUpdateView@http://localhost:4200/vendor.bundle.js:12262:29
callWithDebugContext@http://localhost:4200/vendor.bundle.js:13244:47
detectChanges@http://localhost:4200/vendor.bundle.js:10353:81
activateWith@http://localhost:4200/vendor.bundle.js:21734:55
placeComponentIntoOutlet@http://localhost:4200/vendor.bundle.js:20915:28
activateRoutes@http://localhost:4200/vendor.bundle.js:20896:50
http://localhost:4200/vendor.bundle.js:20832:72
```

**Comment by Sun3** [comment]
ps: For anyone that needs a quick fix without using Two Way data binding I have placed a sample below. For my case this will not work because I have different forms with lot's of financial data fields and it would not be the best. Just the shear number of fields I would need to change.

```
<input type="number" [ngModel]='loanSnapshot?.loanAmount' (ngModelChange)="loanSnapshot.loanAmount = $event">
```

**Comment by jamesamuel** [comment]
@davideast using the .take(1) on a list retrieval no longer works in v4? Why is this, and how does one resolve the issue? i.e. 

`this._db.list('items').take(1).subscribe(res => this.example = res);`

**Comment by markgoho** [comment]
@jamesamuel if you're experiencing a bug, please create a new issue and follow the provided template. This will help us figure out what's going on.

**Comment by funkizer** [comment]
how about wrapping the inputs in something like <div *ngIf="contactObservable | async; let contact"> ... </div>? This way you'll render inputs only after you have data and contact is an object.
